### PR TITLE
Refactor code so that it's typed and has more consistent style

### PR DESCRIPTION
**Pull request by BeebBenjamin**
* Split up component functions into their own classes.
* Added typing and default props.
* Added es-lint and Prettier to catch various errors and typing issues.
* Renamed files to use "Relight" because ARCHiOx is not a functional name.
* Simplified code of the Relight component.
* Added unit tests.
* Fixes to helpers based on testing.

Fixes: #64 #61 #60 #59

### CODE REVIEW

File: src/plugins/RelightHelpers.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
+/**
+ *
+ * @param id
```
Comment (irv) [inline src/plugins/RelightHelpers.js:3]: Feels like we should add some documentation here since we've added the boilerplate

Comment (BeebBenjamin) [reply src/plugins/RelightHelpers.js:3]: I agree.

File: src/plugins/RelightHelpers.js [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        rh = tileHeight;
+        region = 'full';
+      } else {
+        rw = Math.min(regionWidth, width - x);
+        rh = Math.min(regionHeight, height - y);
+        region = x + ',' + y + ',' + rw + ',' + rh;
+      }
+      const scaledWidthRemaining = Math.ceil((width - x) / scale);
+      const tw = Math.min(tileWidth, scaledWidthRemaining);
+      const iiifArgs = '/' + region + '/' + tw + ',/0/default.jpg';
```
Comment (irv) [inline src/plugins/RelightHelpers.js:45]: I think we might need to track rotation here. you can rotate the images using mirador image tools, and then enabling the plugin gives you this: 
image

Comment (irv) [reply src/plugins/RelightHelpers.js:45]: Actually come to think of it, here is not the place to track it. so ignore me and i'll add a seperate issue!

**Pull request merged**